sale_project_service_tracking_copy_tasks/views/product_template_views.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2023 Moduon Team S.L.
+     License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0) -->
+<odoo>
+    <record
+        id="product_template_form_view_service_tracking_copy_tasks_in_project"
+        model="ir.ui.view"
+    >
+        <field
+            name="name"
+        >product.template.service.tracking.copy.tasks.in.project.inherit</field>
+        <field name="model">product.template</field>
+        <field
+            name="inherit_id"
+            ref="sale_project.product_template_form_view_invoice_policy_inherit_sale_project"
+        />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='project_template_id']" position="attributes">
+                <attribute
+                    name="attrs"
+                >{'invisible': [('service_tracking', 'not in', ['task_in_project', 'project_only', 'copy_tasks_in_project'])], 'required': [('service_tracking', 'in', ['copy_tasks_in_project'])]}</attribute>
```

Comment:
**thought(non-blocking):** https://github.com/OCA/server-tools/tree/16.0/base_view_inheritance_extension could let you add this in a more resilient fashion.